s3tests/functional/test_headers.py

```diff
@@ -178,28 +180,18 @@ def wrap(func):
 #
 
 @tag('auth_common')
```

Comment:
@cbodley any idea if there's any use to this tag? If not can we go ahead and remove them in a follow up PR?

Reply:
good question, i left it there because i didn't understand it 🤷 

but looking through the git history, i see that it was originally added in https://github.com/ceph/s3-tests/pull/84, where it was used to filter between auth_common/auth_aws2/auth_aws4:
```
$ S3_USE_SIGV4=1 S3TEST_CONF=your.conf ./virtualenv/bin/nosetests s3tests/functional/test_headers.py -a auth_aws4
```
i'm not sure why the PR added a new `def tag(*tags)` function for this filtering instead of using nose's `@attr()` like we did for fails_on_rgw etc. `@tag` apparently worked with `nosetests -a auth_aws4`, but `pytest -m auth_aws4` doesn't:
```
$ pytest -m auth_aws4
collected 772 items / 772 deselected / 0 selected
```
so i'll change these to pytest markers the same way i did for fails_on_rgw and the others, and drop the `tag()` function

Reply:
The only `tags` I see are `auth_common`, `auth_aws4`, and `auth_aws2`. If we're not using the tags for anything want to just remove them all together?

Reply:
i think it's useful to keep them for filtering. with that latest commit:
```
$ pytest -m auth_aws4
collected 772 items / 735 deselected / 37 selected
s3tests/functional/test_headers.py::test_object_create_bad_md5_invalid_garbage_aws4 ...
```